### PR TITLE
Add swipe refresh to contributors

### PR DESCRIPTION
**Pull request by kobakei**
## Issue
-

## Overview (Required)
- This project is frequently contributed so I added swipe to refresh to contributors page.
- When user swipes to refresh, contributors will be fetched from GitHub API even if data is cached.

## Links
-

## Screenshot
Before | After
:--: | :--:
 | device-2017-02-08-214846

### CODE REVIEW

**Comment by konifar** [comment]
:eyes:

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java [hunk 1]
```diff
@@ -95,13 +95,15 @@ private void initView() {
     }
 
     private void renderContributors(List<ContributorViewModel> contributors) {
-        adapter.addAllWithNotify(contributors);
+        adapter.clear();
+        adapter.addAll(contributors);
+        adapter.notifyDataSetChanged();
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: It's better to use `DiffUtil` to refresh only changed items 🤔 
But no need to fix on this PR!

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java [hunk 2]
```diff
@@ -110,6 +112,17 @@ public void onClickContributor(String htmlUrl) {
...
 
+    @Override
+    public void onSwipeRefresh() {
+        Disposable disposable = viewModel.getContributors(true)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        this::renderContributors,
+                        throwable -> Timber.tag(TAG).e(throwable, "Failed to show contributors.")
+                );
+        compositeDisposable.add(disposable);
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: How about creating private method?

```java
private void showContributors(boolean refresh) {
    Disposable disposable = viewModel.getContributors(true)
        .observeOn(AndroidSchedulers.mainThread())
        .subscribe(
            this::renderContributors,
            throwable -> Timber.tag(TAG).e(throwable, "Failed to show contributors.")
        );
    compositeDisposable.add(disposable);
}
```

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/contributors/ContributorsRepository.java [hunk 3]
```diff
@@ -49,7 +49,7 @@
     }
 
 
-    private Single<List<Contributor>> findAllFromRemote() {
+    public Single<List<Contributor>> findAllFromRemote() {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/contributors/ContributorsRepository.java]: I think it's better to create `setIsDirty(boolean isDirty)` and call it before `findAll()`.
What do you think?

**Comment by konifar** [comment]
Thanks for improvement!
Sorry, can you merge master? 🙇

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java (hunk 1 above)
Comment (kobakei) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: I have not known `DiffUtil`.
I'll try it. Thanks!

**Comment by kobakei** [comment]
OK, I will merge master.

File: app/src/main/java/io/github/droidkaigi/confsched2017/repository/contributors/ContributorsRepository.java (hunk 3 above)
Comment (kobakei) [inline app/src/main/java/io/github/droidkaigi/confsched2017/repository/contributors/ContributorsRepository.java]: Your idea is good. I will modify it.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java (hunk 2 above)
Comment (kobakei) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: Good idea. I will refactor it.

Comment (kobakei) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: I have found another idea to use `ObservableList` in databinding package.
https://developer.android.com/reference/android/databinding/ObservableList.html

I will try either DataUtil or ObservableList on another PR so please ignore this time.

**Comment by kobakei** [comment]
@konifar I have fixed reviewed points!

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java [hunk 4]
```diff
@@ -110,6 +106,21 @@ public void onClickContributor(String htmlUrl) {
         intentOptional.ifPresent(this::startActivity);
     }
 
+    @Override
+    public void onSwipeRefresh() {
+        getContributors(true);
+    }
+
+    private void getContributors(boolean refresh) {
```
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: I think `get~` method name makes us expect some return value.
How about `showContributors(refresh)` or `loadContributors(refresh)`?

Comment (kobakei) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: Okay, fixed.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java (hunk 1 above)
Comment (kobakei) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: I have found another idea to use ObservableList in databinding package.
https://developer.android.com/reference/android/databinding/ObservableList.html

I will try either DataUtil or ObservableList on another PR so please ignore this time.

File: app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java (hunk 2 above)
Comment (konifar) [inline app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/ContributorsFragment.java]: Cool 🆒 I'm looking forward to 😄

**Comment by konifar** [comment]
Perfect 💯 Thanks!!

**Pull request merged**